src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2377,6 +2377,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 {
                     Error(diagnostics, ErrorCode.ERR_RefAssignmentMustHaveIdentityConversion, expression.Syntax, targetType);
                 }
+                else
+                {
+                    return expression;
+                }
```

Comment:
Thought about that. I think as long as we are not allowed to pass tuple literals by ref we are ok. The warning is only reported for tuple literals.

---
In reply to: [81259374](https://github.com/dotnet/roslyn/pull/14176#discussion_r81259374) [](ancestors = 81259374)